### DISCUSSION TITLE
Forced conversation

### DISCUSSION THREAD

**Discussion by MoonshineSG** [python-telegram-bot/python-telegram-bot#2579] [Q&A] [answered]
Is there anyway for a bot to force the ConversationHandler to a certain state for a specific user, bypassing the entry point ?

**Comment by Bibo-Joshi** [comment]
No. Please explain what you want do achieve with this - I'm sure that there is a better solution 😉

**Reply by MoonshineSG** [reply]
There's normal conversation with a CommandHandler entry point and a MessageHandler, but the MessageHandler should handle other cases based on certain criteria... (prob not the best explanation) 

I have a workaround by using `context.user_data` but it's not very elegant. Forcing the conversation would have been my preferred solution. But such is life... For now it works and seams to perform as expected.

**Reply by Bibo-Joshi** [reply]
> prob not the best explanation

yes ;)

> but the MessageHandler should handle other cases based on certain criteria

What criteria are those? When do you want the entry point to trigger, when don't you want it to trigger?

**Reply by MoonshineSG** [reply]
it's based on "business logic". Difficult to explain in short. I'll figure smth else. 

But the question "can ConversationHandler be forced the to a certain state" has been answered, so that can close the case.

**Comment by MoonshineSG** [comment, marked as answer]
in short, the answer is "no"
